src/features/ai/sidebar/SelectionPanelSection.tsx

```diff
@@ -1,52 +1,151 @@
 import { useQuery } from '@tanstack/react-query';
 import { useAtomValue } from 'jotai';
+import { useCallback, useEffect, useState } from 'react';
+import { VscChevronLeft, VscChevronRight } from 'react-icons/vsc';
 import { useDebounce } from 'usehooks-ts';
 
-import { askForRewrite } from '../../../api/rewrite';
+import { askForRewrite, REWRITE_MANNER, RewriteOptions } from '../../../api/rewrite';
 import { selectionAtom } from '../../../atoms/selectionState';
 import { PanelSection } from '../../../components/PanelSection';
 import { emitEvent } from '../../../events';
+import { cx } from '../../../lib/cx';
+import { getCachedSetting } from '../../../settings/settingsManager';
 
 export function SelectionPanelSection() {
   const selection = useAtomValue(selectionAtom);
   const debouncedSelection = useDebounce(selection, 200);
 
+  return (
+    <PanelSection title="Selection">
+      {!selection && <p className="mb-1 mt-4 px-4 italic">Select some text in the editor to see it here.</p>}
+
+      {selection && <RewriteWidget className="mb-6" selection={debouncedSelection} />}
```

Comment:
I think it should be `!debouncedSelection` and `debouncedSelection`, otherwise, when selecting something, we show the widget with empty content during 200ms